nf_core/download.py

```diff
@@ -193,54 +286,101 @@ def prompt_pipeline_name(self):
+                self.revision.append(choice) if isinstance(choice, str) else self.revision.extend(choice)
+            else:
+                if bool(tag_set):
+                    self.revision = tag_set
+                    log.info("No particular revision was selected, all available will be downloaded.")
+                else:
+                    raise AssertionError(f"No revisions of {self.pipeline} available for download.")
 
     def get_revision_hash(self):
         """Find specified revision / branch hash"""
 
-        # Branch
-        if self.revision in self.wf_branches.keys():
-            self.wf_sha = self.wf_branches[self.revision]
-
-        # Revision
-        else:
-            for r in self.wf_revisions:
-                if r["tag_name"] == self.revision:
-                    self.wf_sha = r["tag_sha"]
-                    break
+        for revision in self.revision:  # revision is a list of strings, but may be of length 1
+            # Branch
+            if revision in self.wf_branches.keys():
+                self.wf_sha = {**self.wf_sha, revision: self.wf_branches[revision]}
 
-            # Can't find the revisions or branch - throw an error
+            # Revision
             else:
-                log.info(
-                    "Available {} revisions: '{}'".format(
-                        self.pipeline, "', '".join([r["tag_name"] for r in self.wf_revisions])
+                for r in self.wf_revisions:
+                    if r["tag_name"] == revision:
+                        self.wf_sha = {**self.wf_sha, revision: r["tag_sha"]}
+                        break
+
+                # Can't find the revisions or branch - throw an error
+                else:
+                    log.info(
+                        "Available {} revisions: '{}'".format(
+                            self.pipeline, "', '".join([r["tag_name"] for r in self.wf_revisions])
+                        )
                     )
-                )
-                log.info("Available {} branches: '{}'".format(self.pipeline, "', '".join(self.wf_branches.keys())))
-                raise AssertionError(f"Not able to find revision / branch '{self.revision}' for {self.pipeline}")
+                    log.info("Available {} branches: '{}'".format(self.pipeline, "', '".join(self.wf_branches.keys())))
+                    raise AssertionError(f"Not able to find revision / branch '{revision}' for {self.pipeline}")
 
         # Set the outdir
         if not self.outdir:
-            self.outdir = f"{self.pipeline.replace('/', '-').lower()}-{self.revision}"
-
-        # Set the download URL and return
-        self.wf_download_url = f"https://github.com/{self.pipeline}/archive/{self.wf_sha}.zip"
+            if len(self.wf_sha) > 1:
+                self.outdir = f"{self.pipeline.replace('/', '-').lower()}_{datetime.now().strftime('%Y-%m-%d_%H-%M')}"
```

Comment:
will the datetime not be very similar for multiple revisions? How about using the revision instead?

Reply:
It does this, if only one revision is downloaded, e.g. :
`nf-core-chipseq_2.0.0`
But for more than one revision, it used the date and time of the download instead:
`nf-core-rnaseq_2023-04-21_18-54`

This is particularly important for Tower downloads, which may comprise 10 or more revisions.